### DISCUSSION TITLE
How to invalidate cache when one workspace changes?

### DISCUSSION THREAD

**Discussion by zofiag** [vercel/turborepo#1983] [Help] [answered]
Given monorepo structure
```
- apps
  - web
  - api
- libs
  - ui
- tools
  - eslint-config
```
and `turbo.json` file
```json
{
  "$schema": "https://turborepo.org/schema.json",
   "pipeline": {
    "lint": {
      "dependsOn": ["^lint"],
      "outputs": []
    },
  }
}
```
Now, all the workspaces have `tools/eslint-config` in their dependency list. `apps/web` also has `libs/ui` workspace present under `package.json dependencies`.

What's the best way to invalidate cache for `lint` script in all workspaces every time at least one file under `tools/eslint-config` is changed? In the current setup it works with one side effect that `lint` is rerun in the `apps/web` when any change is introduced in `libs/ui` which is not desired. Is there a way to invalidate workspaces `lint` command only when their own content or `tools/eslint-config` content is changed?

**Comment by nathanhammond** [comment, marked as answer]
This should cover you, possibly with minor tweaks because I'm coding in GitHub:
```
    "lint": {
      "inputs": ["../../tools/eslint-config/**"],
      "outputs": []
    },
```

**Reply by zofiag** [reply]
Thanks! It worked after I upgraded from 1.4.6 to 1.5.6 (https://github.com/vercel/turborepo/pull/1937)
